gpbft/gpbft.go

```diff
@@ -1304,10 +1299,27 @@ func (c *convergeState) Receive(sender ActorID, value ECChain, ticket Ticket, ju
 	c.senders[sender] = struct{}{}
 	key := value.Key()
 
-	// Keep only the first justification and ticket received for a value.
-	if _, found := c.values[key]; !found {
-		c.values[key] = ConvergeValue{Chain: value, Justification: justification}
-		c.tickets[key] = append(c.tickets[key], ConvergeTicket{Sender: sender, Ticket: ticket})
+	senderPower, _ := table.Get(sender)
+	// Keep only the first justification and best ticket
+	if v, found := c.values[key]; !found {
```

Comment:
Do we even need to keep multiple values? Instead, can we just keep the best ticket value (or, possibly, the best ticket for the best candidate)?

Reply:
We could, but that will change if we do the audit recommendation.

Reply:
Wasn't the audit recommendation to skip to the second-best ticket if the best ticket doesn't have a valid candidate? In that case, we can filter candidates up-front. I.e.:

1. Ignore all CONVERGE messages with values that aren't in our candidate set.
2. Take the value with the best ticket from the remaining values.

Reply:
yeah, I guess that could work

Reply:
Quick sanity check: Has the audit recommendation been implemented in this PR?

Reply:
No it has not